### DISCUSSION TITLE
Can we have a tutorial on how to build the code for boosting?

### DISCUSSION THREAD

**Discussion by hasibagen** [Eelbrain/Eelbrain#57] [General]
Can we have a tutorial on how to build the code for boosting?

Thank you very much for developing this package and creating a tutorial on it, it really helped me a lot.

I have read your paper (Brodbeck et al., 2022) carefully and have a preliminary understanding of the theoretical principles of boosting. However, due to my limited programming skills, I may not be able to reproduce the code implementation directly based on the theory idea.
I have read both your paper and [Introduction to Temporal Response Functions (TRFs) - Eelbrain 0.39 documentation](https://eelbrain.readthedocs.io/ en/stable/auto_examples/temporal-response-functions/trf_intro.html#sphx-glr-auto-examples-temporal-response-functions-trf-intro-py ) tutorial. I can understand the principle and code implementation of trf very well.
But the boosting part doesn't seem to have much explanation and tutorials, so I searched the GitHub package myself and only found this (https://github.com/christianbrodbeck/Eelbrain/blob/main/eelbrain/_trf/_boosting_opt.pyx) appears to be the way boosting is implemented, but wasn't quite able to understand the details of it.
So I was wondering if there might be a tutorial on implementing boosting.

Finally, thanks again for your development work. Thank you!

**Comment by christianbrodbeck** [comment]
We have added some material that could be helpful in the [latest version of the preprint](https://www.biorxiv.org/content/10.1101/2021.08.01.454687v3.full.pdf), in particular section 
3.4 Background: The boosting algorithm

**Reply by hasibagen** [reply]
Thank you very, very much indeed, I'm going to go through that part right now.
